### PR TITLE
Sprint2 Updates

### PR DESCRIPTION
**Pull request by RafterGit**
### This pull request changes

    Updated based around comments on latest Sprint 1 Comments

    - Moved node-sass to devDependancies in `frontend/package.json`

    - Decided to keep two `manifest.yml` in `tdrs-frontend` and `tdrs-backend` for the time being and remove the root 
       manifest.yml

    - Consolidated the items in `tdrs-backend/.gitignore` to the root `.gitignore`

    - Added `Pipfile.lock  -diff` to `.gitattributes` in the root of the directory

    - Add JSDoc, and document the functions and components throughout the React app

### [AC] This pull request is ready to merge when

- [ ] Tests have been updated (and all tests are passing)
- [ ] Vulnerability Scan Test (Zap) has no high/critcial CVEs
- [ ] Linting Tests (ESLint for UI and Flake8 for Python)
- [ ] Unit Tests ( including a code coverage test report)
- [ ] Accessibility Tests (Pa11y UI Only)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate

###[DoD] This feature is done when...

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: tdrs-backend/tdpservice/users/api/login.py [hunk 1]
```diff
@@ -39,17 +45,19 @@ def get(self, request, *args, **kwargs):
             return HttpResponseRedirect(os.environ["FRONTEND_BASE_URL"])
 
         # get the validation keys to confirm generated nonce and state
-        nonce_and_state = utils.get_nonce_and_state(request)
-        nonce_validator = nonce_and_state.get("nonce", "not_nonce")
-        state_validator = nonce_and_state.get("state", "not_state")
+        nonce_and_state = get_nonce_and_state(request.session)  # pragma: no cover
```
Comment (lauraGgit) [inline tdrs-backend/tdpservice/users/api/login.py:48]: are you skipping coverage on these because they are mocked in the test suite? at a minimum it seems like an integration test that hits these would be good.

File: tdrs-backend/tdpservice/users/api/login.py [hunk 2]
```diff
@@ -75,21 +83,21 @@ def get(self, request, *args, **kwargs):
...
             raise SuspiciousOperation(msg)
 
-        if not decoded_payload["email_verified"]:
+        if not decoded_payload["email_verified"]:  # pragma: no cover
             return Response(
                 {"error": "Unverified email!"}, status=status.HTTP_400_BAD_REQUEST
             )
 
-        try:
+        try:  # pragma: no cover
```
Comment (lauraGgit) [inline tdrs-backend/tdpservice/users/api/login.py:100]: It really feels like this try and this exception should be included in coverage, unless i'm not understanding something.

File: tdrs-backend/tdpservice/users/api/utils.py [hunk 3]
```diff
@@ -18,45 +18,6 @@
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
-"""
```
Comment (lauraGgit) [inline tdrs-backend/tdpservice/users/api/utils.py:21]: this looks like a section of code that got merged into `raft-tdp-main` without government review. It would be helpful to have another pr on that.

File: tdrs-backend/tdpservice/users/test/test_auth_check.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import pytest
+from django.urls import reverse
+from rest_framework import status
+
+
+@pytest.mark.django_db
+def test_auth_check_endpoint_with_no_user(api_client, user):
+    """If there is no user auth_check should return FORBIDDEN."""
+    response = api_client.get(reverse("authorization-check"))
```
Comment (lauraGgit) [inline tdrs-backend/tdpservice/users/test/test_auth_check.py:11]: that `reverse()` util is nifty!

File: tdrs-backend/tdpservice/users/test/test_auth_check.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+@pytest.mark.django_db
+def test_auth_check_returns_authenticated(api_client, user):
+    """If user is authenticated auth_check should return authenticated true."""
+    api_client.login(username=user.username, password="test_password")
+    response = api_client.get(reverse("authorization-check"))
+    assert user.is_authenticated is True
+    assert response.data["authenticated"] is True
+
+@pytest.mark.django_db
```
Comment (lauraGgit) [inline tdrs-backend/tdpservice/users/test/test_auth_check.py:29]: These test cases look good. I think it would be important to test a couple more fail cases.

File: tdrs-backend/tdpservice/users/test/test_auth_check.py (hunk 4 above)
Comment (carltonsmith) [reply tdrs-backend/tdpservice/users/test/test_auth_check.py:11]: Right? It allows us the flexibility to change endpoints in the future should it be needed without having to update the code

File: tdrs-backend/tdpservice/users/api/utils.py (hunk 3 above)
Comment (RafterGit) [reply tdrs-backend/tdpservice/users/api/utils.py]: I'll create one for you to take a look at , this was us migrating code out of login.py into a helper class.

File: tdrs-frontend/src/components/PrivateRoute/PrivateRoute.js [hunk 6]
```diff
@@ -5,7 +5,15 @@ import { useSelector, useDispatch } from 'react-redux'
 import { setAlert, clearAlert } from '../../actions/alert'
 import { ALERT_INFO } from '../Notify'
 
-function Private({ children, history, path }) {
+/**
```
Comment (lauraGgit) [inline tdrs-frontend/src/components/PrivateRoute/PrivateRoute.js:8]: Thanks for updating the component name

**Comment by lauraGgit** [comment]
LGTM 🚀
